### PR TITLE
Manual buffer management in restore-tar

### PR DESCRIPTION
**Pull request by tgruben**
fixes #fb-1912

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -10,9 +10,7 @@ run:
     - pql/pql.peg.go
 linters:
   enable:
-    - govet
-    - gofmt
-    - staticcheck
+    - gosimple
```
Comment (seebs) [inline .golangci.yml]: Is this change intentional? It seems surprising to me because I was pretty sure we did in fact want those three, especially staticcheck.

File: ctl/multi-reader.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package ctl
+
+// Reader wraps an io.Reader, allowing "clones" to be created via the View method.
+import (
+	"bytes"
+	"io"
+	"sync"
+)
+
+type MultiReader struct {
```
Comment (seebs) [inline ctl/multi-reader.go]: I think we should pick a different name for this, because `io.MultiReader` already exists, for a reader which combines several other readers. Maybe `ReReader`?

**Review by seebs — DISMISSED** [review]
so this looks great, but we should probably squash the commits and clean up the commit message? but otherwise this looks pretty reasonable. we do need better testing on this but you're already on that.

**Comment by tgruben** [comment]
thx @seebs i'll squash after these test clear

**Review by seebs — APPROVED** [review]
i think there's a typo in an error message (`tran&slate`) but this otherwise looks good to me.

i particularly like removing the `// this feels wrong but works for now` comment on a thing that was apparently not working well enough. :)

**Review by seebs — APPROVED** [review]
fixed typo looks good!
